### ISSUE TITLE
POST with URI encoded data instead of JSON

### ISSUE THREAD

**Issue by guoyunhe** [suhaotian/xior#26]
There are some really old back-end systems that don't support JSON API. Instead, they support encoded URI data as body.

axios supports a usage like this:

```js
axios.post('/foobar', new URLSearchParams({
  foo: 'bar',
}));
```

axios will automatically set `content-type` header to `application/x-www-form-urlencoded;charset=UTF-8`;

However, xior doesn't seem to support this. When I write:

```js
xior.post('/foobar', new URLSearchParams({
  foo: 'bar',
}));
```

xior will send a request with no `content-type` header, and back-end will reject it.

I can fix my code by manually adding the missing header. But maybe it makes sense that xior can do this automatically for users.

**Comment by suhaotian** [comment]
Good catch! 👊👍 I will enhance it in next version.

---

Currently, there is a better way to do that:

```ts
import xiorInstance from 'xior';

xiorInstance.defaults.headers['content-type'] = `application/x-www-form-urlencoded`;


xior.post('/foobar', {foo: 'bar'});
```

**Comment by suhaotian** [comment]
Fixed in `v0.5.4` 🥳  now support:

```ts
xior.post('/', new URLSearchParams({foo: 'bar'}))
```
